tools/publisher/src/subcommand/publish.rs

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use crate::cargo;
+use crate::fs::Fs;
+use crate::package::{discover_package_batches, PackageBatch};
+use crate::repo::discover_repository;
+use crate::{REPO_CRATE_PATH, REPO_NAME};
+use anyhow::Result;
+use dialoguer::Confirm;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::sync::Semaphore;
+use tracing::info;
+
+const BACKOFF: Duration = Duration::from_millis(30);
+const MAX_CONCURRENCY: usize = 4;
```

Comment:
[The actual max is 30](https://github.com/rust-lang/crates.io/blob/d7dad029e7dd582cc26fb3daae50091af574cdbd/config/nginx.conf.erb#L268) afaict

https://github.com/rust-lang/crates.io/issues/3229